src/client/pages/SiaDetail/SiaDetail.tsx

```diff
@@ -0,0 +1,341 @@
+import { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import { atom, RecoilState, useRecoilState } from 'recoil';
+import {
+  SiaAttachment,
+  SIAItem,
+  SiaSignalStatusHistory,
+} from '../../../server/services/sia';
+import { AppRoutes, ChapterTitles } from '../../../universal/config';
+import {
+  apiPristineResult,
+  ApiResponse,
+  defaultDateFormat,
+  isError,
+  isLoading,
+} from '../../../universal/helpers';
+import {
+  Alert,
+  ChapterIcon,
+  DetailPage,
+  DocumentList,
+  LoadingContent,
+  PageContent,
+  PageHeading,
+} from '../../components';
+import { LinkdInline } from '../../components/Button/Button';
+import InfoDetail, {
+  InfoDetailGroup,
+} from '../../components/InfoDetail/InfoDetail';
+import { TextClamp } from '../../components/InfoDetail/TextClamp';
+import StatusLine from '../../components/StatusLine/StatusLine';
+import { BFF_API_BASE_URL } from '../../config/api';
+import { useAppStateGetter } from '../../hooks/useAppState';
+import { Location } from '../VergunningDetail/Location';
+import styles from './SiaDetail.module.scss';
+
+function getSiaMeldingStatusLineItems(
+  SiaItem: SIAItem | undefined,
+  history: ApiResponse<SiaSignalStatusHistory[]>
+) {
+  if (!SiaItem) {
+    return [];
+  }
+
+  const statusLineItems = (history.content ?? []).map(
+    (historyItem, index, all) => {
+      return {
+        id: 'item-' + historyItem.key,
+        status: historyItem.status,
+        datePublished: historyItem.datePublished,
+        description: historyItem.description,
+        documents: [],
+        isActive: index === all.length - 1,
+        isChecked: true,
+      };
+    }
+  );
+
+  // Add "dummy" statuses so we can show a process flow.
+  if (!statusLineItems.length) {
+    statusLineItems.push({
+      id: 'item-1',
+      status: 'Open',
+      datePublished: SiaItem.datePublished,
+      description: '',
+      documents: [],
+      isActive: true,
+      isChecked: true,
+    });
+  }
+  if (
+    statusLineItems.length &&
+    statusLineItems[statusLineItems.length - 1].status !== 'Afgesloten'
+  ) {
+    statusLineItems.push({
+      id: 'item-3',
+      status: 'Afgesloten',
+      datePublished: '',
+      description: '',
+      documents: [],
+      isActive: false,
+      isChecked: false,
+    });
+  }
+
+  return statusLineItems;
+}
+
+const statusHistoryAtom = atom<
+  Record<string, ApiResponse<SiaSignalStatusHistory[]>>
+>({
+  key: 'siaStatusHistoryAtom',
+  default: {},
+});
+
+const attachmentsAtom = atom<Record<string, ApiResponse<SiaAttachment[]>>>({
+  key: 'siaAttachmentsAtom',
+  default: {},
+});
+
+function useAdditionalDataById<T extends ApiResponse<any>>(
+  id: string,
+  url: string,
+  atom: RecoilState<Record<string, T>>
+): T {
+  const [data, setData] = useRecoilState(atom);
+  const isDataFetched = id in data;
+
+  useEffect(() => {
```

Comment:
Blijft de call naar `setData` goed gaan als het component dat de hook aanroept ge-unmount wordt?

Reply:
Het effect wordt sowieso uitgevoerd en mits een Ok response wordt de Recoil Atom gevuld met data. Zelfs als er in de tussentijd een unmount is. De fetch data call is a-synchroon en wordt niet expliciet gecancelled of opgeruimd. Omdat de Recoil state niet lokaal is kan de data in een volgende navigatie naar detail pagina direct gebruikt worden.

Daarnaast zie ik niet direct grote nadelige gevolgen van het opslaan van opgehaalde data die eventueel niet meer gebruikt wordt indien een gebruiker niet nogmaals de detailpagina aanroept. 

Kun jij wat bedenken?